src/mapnik_vector_tile_simple_valid.cpp

```diff
@@ -429,13 +417,13 @@ void layer_not_valid(protozero::pbf_reader& layer_msg,
             bool has_geom = false;
             bool has_geom_type = false;
             std::int32_t geom_type_enum = 0;
-            std::uint64_t feature_id = 0;
+            //std::uint64_t feature_id = 0;
             while (feature_msg.next())
             {
                 switch (feature_msg.tag())
                 {
                 case mapnik::vector_tile_impl::Feature_Encoding::ID:
-                    feature_id = feature_msg.get_uint64();
+                    /*feature_id*/ feature_msg.get_uint64();
```

Comment:
perhaps we are just missing test coverage? This seems dangerous to disable...

Reply:
as far as I know, compiler won't ever optimise function call away if a return value is ignored. But it'll be better to remove `case mapnik::vector_tile_impl::Feature_Encoding::ID:` completely, provided `protozero` can handle that.

Reply:
And yes it can, this is what `feature_msg.skip();` for, going to amend ..